src/sentry/auth/access.py

```diff
@@ -340,7 +371,13 @@ def from_member(
     # network hops and needed in a lot of places
     requires_sso, sso_is_valid = _sso_params(member)
 
-    team_list = frozenset(member.get_teams())
+    team_memberships = frozenset(
+        TeamMembership.from_team_member(omt)
+        for omt in OrganizationMemberTeam.objects.filter(organizationmember=member, is_active=True)
+        if omt.team.status == TeamStatus.VISIBLE
```

Comment:
I'm not sure how much `omt.team.status == TeamStatus.VISIBLE` actually matters, but I included it to faithfully replicate the behavior of `OrganizationMember.get_teams()`. I added `test_excludes_invisible_teams` to test it.

If team status is significant, we might also want to add it to the "all teams" lookup under `_from_sentry_app`.